proof-libs/fstar/core/Rand_core.fsti

```diff
@@ -0,0 +1,12 @@
+module Rand_core
```

Comment:
This is not part of the Rust core library (or std or alloc). So it probably shouldn't be in here.

std and alloc should go into their own folders too. But that may be ok for now.

Reply:
Merging this for now, and adding an issue to clean up this folder structure.